### ISSUE TITLE
Markdown: meta with wikilinks dosn't work

### ISSUE THREAD

**Issue by tkliuxing** [django-wiki/django-wiki#464]
### Test case

I use `testproject` test this.

In `testproject/settings/local.py`: 

```python
WIKI_MARKDOWN_KWARGS = {
    'extensions': [
        'meta',
        'wikilinks',
        'codehilite',
        'footnotes',
        'attr_list',
        'headerid',
        'extra',
    ], 'safe_mode': False}
```

wiki markdown content:

```markdown
wiki_base_url: /prefixurl/

This is [[Link]]
```

According to http://pythonhosted.org/Markdown/extensions/wikilinks.html
It should be:
```html
<p>This is <a href="http://somewhere/prefixurl/Link" class="wikilink">Link</a></p>
```

But `wiki_base_url` dons't add to url:
```html
<p>This is <a href="http://somewhere/Link" class="wikilink">Link</a></p>
```

### My environment

OS: _OSX 10.10.5_
python: _2.7.10_

pip freeze:
```
appnope==0.1.0
decorator==4.0.2
Django==1.8
django-classy-tags==0.6.2
django-mptt==0.7.1
django-nyt==0.9.7.2
django-sekizai==0.8.2
gnureadline==6.3.3
ipython==4.0.0
ipython-genutils==0.1.0
Markdown==2.6.2
path.py==8.1
pexpect==3.3
pickleshare==0.5
Pillow==2.9.0
simplegeneric==0.8.1
six==1.9.0
sorl-thumbnail==12.3
South==1.0.2
traitlets==4.0.0
wheel==0.24.0
-e git+git@github.com:tkliuxing/django-wiki.git@d14e6b86f3e5f3a47d41358830cc806dc12b8b07#egg=wiki-base_master
```

**Comment by benjaoming** [comment]
From [the docs](http://pythonhosted.org/Markdown/extensions/wikilinks.html):

> base_url: String to append to beginning of URL.

How about that? :)

**Comment by tkliuxing** [comment]
See this docs part: [Using with Meta-Data extension](http://pythonhosted.org/Markdown/extensions/wikilinks.html#using-with-meta-data-extension)


I tried the method you said, it also doesn't work, but `wiki_end_url` meta attribute work.
Like this article:
```markdown
wiki_end_url: .html

This is [[Link]]
```

Output:
```html
<p>This is <a href="http://somewhere/Link.html"></a></p>
```

**Comment by benjaoming** [comment]
I can't really help with this WikiLinks is a part of Markdown, not django-wiki. And using the WikiLink structure isn't really a Markdown-like construction that I really wanna recommend :)

**Comment by tkliuxing** [comment]
@benjaoming I found where is problem: https://github.com/django-wiki/django-wiki/blob/master/wiki/plugins/macros/mdx/wikilinks.py#L18 It replace `base_url` to `/` !

**Comment by benjaoming** [comment]
Okay I'm not actually sure what that module is responsible for :) What happens if you use the original Markdown extension?

**Comment by tkliuxing** [comment]
When I removed macro plugin: 
```python
INSTALLED_APPS = [
    'django.contrib.humanize',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.sites',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.admin',
    'django.contrib.admindocs',
    'sekizai',
    'sorl.thumbnail',
    'django_nyt',
    'wiki',
    # 'wiki.plugins.macros',
    'wiki.plugins.help',
    'wiki.plugins.links',
    'wiki.plugins.images',
    'wiki.plugins.attachments',
    'wiki.plugins.notifications',
    'mptt',
]
```

`wiki_base_url: /Test/` worked well.

**Comment by tkliuxing** [comment]
original Markdown extension test:
```python
import markdown
text = """wiki_base_url: /somewhere/
# This is Title
This is [[Links]]
"""
html = markdown.markdown(text, extensions=['meta','wikilinks','extra'])
print(html)
```
output:
```html
<h1>This is Title</h1>
<p>This is <a class="wikilink" href="/somewhere/Links/">Links</a></p>
```

**Comment by jtebert** [comment]
I'm confused on this. What is recommended for django-wiki to link between pages on the wiki? Is it to use the `[[ Link ]]` syntax? If so, is there a resolution to the problem that it doesn't respect prefix URLs? I have my wiki under the `/wiki/` URL, so I need a way to do this.

**Comment by benjaoming** [comment]
Would you be able to test the proposed fix?

**Comment by jtebert** [comment]
The suggestion of removing the macros plugin? Commenting that out made both [TOC] and wikilinks stop working completely. I haven't been able to successfully get a prefix URL to work at all.

**Comment by benjaoming** [comment]
@jtebert no the PR

**Comment by jtebert** [comment]
PR?

**Comment by benjaoming** [comment]
PR=Pull Request, the link above refers to a branch in Git that I've implemented a fix in. If you're able to run that branch on your system, you can see if it works.

**Comment by jtebert** [comment]
I feel like an idiot, but I don't use any of Git/Github's more complex features so.... how do I actually test the pull request? (Currently using the version from pip)

**Comment by Russell-Jones** [comment]
Just in case you've not already sorted this, you can test it in a
virtualenv.
First set up and activate the VE, then run (if this is the right Julia
Ebert)
pip install --upgrade -e git+
https://github.com/jtebert/django-wiki.git#egg=wiki
and it'll go in /path/to/ve/src/
IIRC, you get prompted for username and password if needed.

Russell

**Comment by benjaoming** [comment]
Presumed fixed in #473
